internal/include/go_context.h

```diff
@@ -72,20 +79,82 @@ static __always_inline struct span_context *get_parent_span_context(void *ctx) {
     return parent_sc;
 }
 
-static __always_inline void start_tracking_span(void *ctx, struct span_context *sc) {
-    bpf_map_update_elem(&tracked_spans, &ctx, sc, BPF_ANY);
-    bpf_map_update_elem(&tracked_spans_by_sc, sc, &ctx, BPF_ANY);
+static __always_inline void start_tracking_span(void *contextContext, struct span_context *sc) {
+    long err = 0;
+    err = bpf_map_update_elem(&tracked_spans, &contextContext, sc, BPF_ANY);
+    if (err != 0)
+    {
+        bpf_printk("Failed to update tracked_spans map: %ld", err);
+        return;
+    }
+
+    err = bpf_map_update_elem(&tracked_spans_by_sc, sc, &contextContext, BPF_ANY);
+    if (err != 0)
+    {
+        bpf_printk("Failed to update tracked_spans_by_sc map: %ld", err);
+        return;
+    }
 }
 
-static __always_inline void stop_tracking_span(struct span_context *sc) {
+static __always_inline void stop_tracking_span(struct span_context *sc, struct span_context *psc) {
+    if (psc == NULL || sc == NULL) {
```

Comment:
Isn't NULL psc a valid state? If we are now creating the root span psc will be NULL

Reply:
Currently this function is passed with pointers to the stack. A root span will have a valid pointer but psc will be zero, or not existing in the map. I changed the logic here to fit other scenarios as well.